### PR TITLE
Enable per-head gradients

### PR DESCRIPTION
**Pull request by luciaquirke**
- Fix bug where projection device was not correct for multi-GPU training setups
- Enable per-head gradients in the callback and static build
- Add some documentation to the README

Notes
- Could add h.*.attn... formatting

### CODE REVIEW

**Review by norabelrose — APPROVED** [review]
LGTM thanks
